Upstream transform streams tests

ricea opened the pull request:
Copy the tests for transform streams from the whatwg/streams repository.

TransformStream has now been added to the Streams Standard so it seems
reasonable to move these tests to the web-platform-tests repository. A follow-on
change will remove them from the whatwg/streams repository.

See https://github.com/whatwg/streams/issues/851.

There are no changes to the test Javascript files. They have all been
reviewed in their old location. All the HTML files have been regenerated by
generate-test-wrappers.js.

recording-streams.js has an additional "recordingTransformStream" function which
is used in some of the new tests. test-utils.js has an additional
"constructorThrowsForAll" test helper.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

ricea committed:
Upstream transform streams tests

Copy the tests for transform streams from the whatwg/streams repository.

TransformStream has now been added to the Streams Standard so it seems
reasonable to move these tests to the web-platform-tests repository. A follow-on
change will remove them from the whatwg/streams repository.

There are no changes to the test Javascript files. They have all been
reviewed in their old location. All the HTML files have been regenerated by
generate-test-wrappers.js.

recording-streams.js has an additional "recordingTransformStream" function which
is used in some of the new tests. test-utils.js has an additional
"constructorThrowsForAll" test helper.

ricea committed:
Add missing test to patched-global.js

The test added in https://github.com/whatwg/streams/pull/852 was missing
from the previous commit. Add it.

w3c-bots commented:
# Build PENDING

Started: None
Finished: None


<h2>Failing Jobs</h2>
<ul>

<li>chrome:unstable</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20258)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/298518615)

ricea commented:
Failed due to lint errors:

streams/transform-streams/general.js:125: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:125: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
streams/transform-streams/general.js:147: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:147: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
streams/transform-streams/general.js:148: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:148: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
streams/transform-streams/general.js:244: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:244: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
streams/transform-streams/general.js:245: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:245: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
streams/transform-streams/general.js:246: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)
ERROR:lint:streams/transform-streams/general.js:246: setTimeout used; step_timeout should typically be used instead (SET TIMEOUT)

I think I will probably fix this over at whatwg/streams so we can review the fix in isolation, then merge it back here.

domenic commented:
Hmm, I think I'd rather fix them here, or at least ensure we don't merge to whatwg/streams before verifying here that they pass lint.

Anyway, I'll approve this now so that when we do get lint passing, we can merge regardless of timezones. The substance of these have all seen review over in whatwg/streams.

ricea committed:
Replace calls to setTimeout with delay()

ricea commented:
I did the minimal change (replacing setTimeout with delay), so we can proceed with upstreaming the tests.

Once we've done the move, I'd like to follow-up with rewriting those tests to not use non-zero delays.

ricea commented:
The Firefox Nightly stability test is failing on Travis. It looks like an issue with the Firefox nightly rather than the tests themselves.

ricea commented:
The Firefox Nightly stability test is still broken today. I will try to get it working on my own machine.

ricea commented:
I've looked at the other stability tests. Safari and Edge are both failing with infrastructure errors. The Chrome test is complaining that lipfuzz.html has an error. But it just reports "TransformStream is not defined" like all the other tests. From a stability point of view, it's 100% stable.

My theory for what is happening with the Firefox bot is that the sheer volume of errors it produces is hitting some limit which prevents the test from passing. I could probably mitigate this by splitting into one PR per file, but I'd rather not do that. It there someone I can ask to get an override for the CI build failure?

ricea committed:
Fix duplicate test name in lipfuzz.js

Because the stringification of [] and [''] is the same, the test name
"testing """ was used twice. Include input array length in the test name
to make them unique.

ricea commented:
Okay, so there was a problem with lipfuzz.js after all.
